src/probes/telem_record_gen.c

```diff
@@ -322,31 +332,59 @@ int get_payload(char **payload)
         return ret;
 }
 
-int send_record(char *payload)
+int instanciate_record(struct telem_ref **t_ref, char *payload)
 {
-        struct telem_ref *t_ref;
         int ret = 0;
 
-        if (tm_create_record(&t_ref, (uint32_t)severity,
-                             opt_class, payload_version) < 0) {
+        if ((ret = tm_create_record(t_ref, (uint32_t)severity,
+                                    opt_class, payload_version)) < 0) {
                 goto out1;
         }
 
-        if (opt_event_id && tm_set_event_id(t_ref, opt_event_id) < 0) {
-                goto out2;
+        if ((ret = opt_event_id && tm_set_event_id(*t_ref, opt_event_id)) < 0) {
+                goto out1;
         }
 
-        if (tm_set_payload(t_ref, payload) < 0) {
-                goto out2;
+        if ((ret = tm_set_payload(*t_ref, payload)) < 0) {
+                goto out1;
         }
+out1:
+        return ret;
+}
 
-        if (tm_send_record(t_ref) >= 0) {
-                ret = 1;
+int send_record(char *payload)
+{
+        struct telem_ref *t_ref = NULL;
+        int ret = 0;
+
+        if ((ret = instanciate_record(&t_ref, payload)) < 0) {
+                goto out;
+        }
+
+        if ((ret = tm_send_record(t_ref)) < 0) {
+                goto out;
         }
 
-out2:
+out:
         tm_free_record(t_ref);
-out1:
+
+        return ret;
+}
+
+int print_record(char *payload)
+{
+        struct telem_ref *t_ref;
```

Comment:
Should initialize this to NULL so that the `tm_free_record()` call below is safe.